src/router.js

```diff
@@ -70,7 +70,7 @@ export function LocationProvider(props) {
 
 	const value = useMemo(() => {
 		const u = new URL(url, location.origin);
-		const path = u.pathname.replace(/(.)\/$/g, '$1');
+		const path = u.pathname.replace(/\/+$/g, '') || '/';
```

Comment:
```suggestion
		const path = u.pathname.replace(/\/+$/g, '/');
```

Reply:
I aimed for maintaining the logic which was ensuring at least a '/' was returned and a single '/' is not stripped.

Reply:
I thought about it as well and following your arguments of being in line with the "standard" in https://github.com/preactjs/preact-iso/pull/27#issuecomment-2187420265 I'd opt for dropping this tinkering with the path completely.

Reply:
> ensuring at least a '/' was returned and a single '/' is not stripped.

Replacing one or more trailing slashes with `/` should do this though? Unless I'm missing something (and I very well might be, let me know!) these are equivalent, just with my suggestion avoiding a fallback.

 > I thought about it as well and following your arguments of being in line with the "standard" in https://github.com/preactjs/preact-iso/pull/27#issuecomment-2187420265 I'd opt for dropping this tinkering with the path completely.

I'm not against landing something here for now, incremental improvements are still very useful and all, and we will need to make adjustments in a major to be far more sensitive with trailing slashes to support this anyhow:

```js
const pattern = new URLPattern("/books/*", "https://example.com");
console.log(pattern.test("https://example.com/books")); // false
console.log(pattern.test("https://example.com/books/")); // true
```

I get the logic, but man does this feel like a footgun in the spec 😅

Reply:
> > ensuring at least a '/' was returned and a single '/' is not stripped.
> 
> Replacing one or more trailing slashes with `/` should do this though? Unless I'm missing something (and I very well might be, let me know!) these are equivalent, just with my suggestion avoiding a fallback.

Reading my reply I'm a bit confused as well ...

As I understand the current code: at the end of the path a single '/' is stripped. If the path is just '/' this stays '/'.

`/my/path/` -> `/my/path`
`/` -> `/`

The aim of my changes was to strip all slashes at the end:

`/my/path/////` -> `/my/path`

but keep:
`/` -> `/`

Yours changing the current behavior to maintain a single slash at the end (if present). I'm fine with that was well. Or we just drop the whole slash-stripping and return the path as it is and leave it to the user to decide how to deal with trailing slashes. In the meantime this is my favorite option ...

> ```js
> const pattern = new URLPattern("/books/*", "https://example.com");
> console.log(pattern.test("https://example.com/books")); // false
> console.log(pattern.test("https://example.com/books/")); // true
> ```
> 
> I get the logic, but man does this feel like a footgun in the spec 😅

Indeed. Isn't this denying the current state that e.g. `https://github.com/preactjs/preact-iso/` returns the same result as `https://github.com/preactjs/preact-iso`?

Reply:
> As I understand the current code: at the end of the path a single '/' is stripped. If the path is just '/' this stays '/'.

Oh yup, you're absolutely right. Sorry, and thanks!

> Indeed. Isn't this denying the current state that e.g. https://github.com/preactjs/preact-iso/ returns the same result as https://github.com/preactjs/preact-iso?

Not denying per se, but (for some reason that I still can't discern), they want users to be able to distinguish against `/preact-iso` & `/preact-iso/`. I don't quite understand who needs this & why though.

It's not too much work for the lib to add an optional trailing slash to bare patterns though:

```js
const pattern = new URLPattern({ pathname: "some-path{/*}?" })
```

but still weird to distinguish, IMO.

---

As for this issue, sorry it's been left open for so long.

I'm going to try to sit down later today to fully map out where we are & where we want to get to, so that I can then figure out the intermediary steps to get there w/ minimal changes.

I appreciate you bearing with me.

Reply:
> As for this issue, sorry it's been left open for so long.
>
> I'm going to try to sit down later today to fully map out where we are & where we want to get to, so that I can then figure out the intermediary steps to get there w/ minimal changes.

No worries about that. Better to have a good solution than a quick one. There's no crucial functionality depending on this PR. In my opinion, it can mature until it is ready.
 
> I appreciate you bearing with me.

Same here :)